binzx/otomi

```diff
@@ -0,0 +1,243 @@
+  last_file_change=$(date -u -r $script_full_path '+%Y-%m-%dT%H:%M:%SZ')
+  last_commit_date_sec=$(date_to_format $last_commit_date)
+  last_file_change_sec=$(date_to_format $last_file_change)
+
+  if [ "$last_commit_date_sec" -ne "0" ] && [ "$last_commit_date_sec" -gt "$last_file_change_sec" ]; then
+    read -r -p "Newer version is available, do you want to update (yes/No)? " answer
+    case ${answer:0:1} in
+    y | Y)
+      run_self_update
+      ;;
+    *)
+      return 0
+      ;;
+    esac
+  fi
+}
+silent echo "Checking for updates"
+[ -z $CI ] && check_update
+
+silent echo "Preparing docker environment variables"
+tmp_env=$(mktemp)
+
+cat >>$tmp_env <<EOF
+ENV_DIR=${ENV_DIR}
+AZURE_TENANT_ID=${AZURE_TENANT_ID}
+AZURE_CLIENT_ID=${AZURE_CLIENT_ID}
+AZURE_CLIENT_SECRET=${AZURE_CLIENT_SECRET}
+AWS_REGION=${AWS_REGION}
+AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+VAULT_TOKEN=${VAULT_TOKEN}
+GCLOUD_SERVICE_KEY=${GCLOUD_SERVICE_KEY}
+K8S_CONTEXT=${K8S_CONTEXT}
+SHELL=${path_to_exec_shell}
+OTOMI_TAG=${OTOMI_TAG}
+OTOMI_CALLER_COMMAND=${BASH_SOURCE[0]##*/}
+OTOMI_IN_DOCKER=true
+EOF
+
+helm_config="$HOME/.config/helm"
+uname -a | grep -i darwin >/dev/null && helm_config="$HOME/Library/Preferences/helm"
+
+stack_dir='/home/app/stack'
+executable="node --experimental-specifier-resolution=node ${stack_dir}/dist/otomi.js --"
+
+stack_volume=""
+tmp_volume_dir=$(mktemp -d)
+tmp_volume=" -v $tmp_volume_dir:/tmp"
+if [[ $base_dir == *"otomi-core"* ]] || [[ $(pwd) == *"otomi-core"* ]]; then
+  silent echo "Checking if in dev environment"
+  if [[ $base_dir == *"otomi-core"* ]]; then
+    stack_dir=$(cd "$base_dir/.." && pwd)
+  elif [[ $(pwd) == *"otomi-core"* ]]; then
+    stack_dir=$(pwd | awk -F'otomi-core' '{print $1"otomi-core"}')
+  fi
+
+  stack_volume="-v $stack_dir:$stack_dir"
+  tmp_volume="-v /tmp:/tmp"
+
+  # Use development distribution if exists
+  if [ -f "${stack_dir}/dist/otomi.js" ]; then
+    executable="node --experimental-specifier-resolution=node ${stack_dir}/dist/otomi.js --"
+  fi
+  echo "OTOMI_DEV=true" >>$tmp_env
+fi
+
+check_volume_path() {
+  source=$1
+  dest=$2
+  [ -z $source ] && [ -z $dest ] && echo "" && return 0
+  echo "-v $source:$dest"
+  return 0
+}
+
+cmd="${executable} $*"
+# If command is "otomi bash"
+if [ "$1" = "bash" ] && [ "$#" = "1" ]; then
+  cmd="bash"
+fi
+
+silent echo "Pulling latest version of the docker image"
+# TODO: Remove otomi-zx check
+[ "$OTOMI_TAG" = 'latest' ] || [ "$OTOMI_TAG" = 'master' ] || [ "$OTOMI_TAG" = 'otomi-zx' ] && [ -z $NOPULL ] && silent docker pull $otomi_tools_image
+status=$?
+if [ "$status" -ne 0 ]; then
+  echo "Could not find an image for the supplied OTOMI_TAG: '${OTOMI_TAG}'"
+  exit $status
+fi
+mkdir -p /tmp/otomi
+docker run --network host --rm -it \
```

Comment:
Previously I could run a command inside the otomi/core container with the DinD flag, would it be possible to reintroduce that functionality?

Reply:
You mean `otomi bash` or `otomi x`?

Reply:
Yes, maybe it is too much of an edge case, but suppose I am in `otomi bash` and I want to run `binzx/otomi`. I now got it working by hacking my way out of the messages like "must specify an ENV_DIR" and "cannot run dist/otomi.js without the binzx/otomi entrypoint".

Reply:
As discussed, if you need to run `./dist/otomi.js` you need a couple of volumes & env variables, as set in `binzx/otomi`